packages/utils/src/conversions.ts

```diff
@@ -0,0 +1,202 @@
+ */
+export function bigIntToHexadecimal(n: bigint): string {
+    return beBigIntToHexadecimal(n)
+}
+
+/**
+ * Converts a hexadecimal string to a bigint, ensuring the hex string starts with '0x'.
+ * @param hex The hexadecimal string to convert.
+ * @returns The bigint representation of the hexadecimal string.
+ */
+export function beHexadecimalToBigint(hex: string): bigint {
+    // Ensure the hex string starts with '0x'.
+    const formattedHexString = hex.startsWith("0x") ? hex : `0x${hex}`
+
+    return BigInt(formattedHexString)
+}
+
+/**
+ * Converts a hexadecimal string to a bigint. Alias for beHexadecimalToBigint.
+ * @param hex The hexadecimal string to convert.
+ * @returns The bigint representation of the hexadecimal string.
+ */
+export function hexadecimalToBigint(hex: string): bigint {
+    return beHexadecimalToBigint(hex)
+}
+
+/**
+ * Converts a buffer to a bigint by interpreting the buffer as a hexadecimal string.
+ * @param b The buffer to convert.
+ * @returns The bigint representation of the buffer's contents.
+ */
+export function beBufferToBigint(b: Buffer): bigint {
+    return BigInt(`0x${b.toString("hex")}`)
+}
+
+/**
+ * Converts a buffer to a bigint. Alias for beBufferToBigint.
+ * @param b - The buffer to convert.
+ * @returns The bigint representation of the buffer's contents.
+ */
+export function bufferToBigint(b: Buffer): bigint {
+    return beBufferToBigint(b)
+}
+
+/**
+ * Converts a bigint to a buffer, assuming big-endian byte order, and fills with zeros if necessary.
+ * @param n The bigint to convert.
+ * @param length The number of bytes of the buffer to return.
+ * @returns The buffer representation of the bigint.
+ */
+export function beBigintToBuffer(n: bigint, length?: number): Buffer {
+    const hex = bigIntToHexadecimal(n)
+
+    // Calculate the minimum buffer length required to represent 'n' in bytes.
+    // Each hexadecimal character represents 4 bits, so 2 characters are 1 byte.
+    const minLength = Math.ceil(hex.length / 2)
+
+    // Use the provided length or the calculated minimum length, whichever is greater.
+    const bufferLength = length ? Math.max(length, minLength) : minLength
+
+    // Allocate buffer of the desired size, filled with zeros.
+    const buffer = Buffer.alloc(bufferLength, 0)
+
+    const fromHex = Buffer.from(hex, "hex")
+    fromHex.copy(buffer, bufferLength - fromHex.length)
+
+    return buffer
+}
+
+/**
+ * Converts a bigint to a buffer. Alias for beBigintToBuffer.
+ * @param n - The bigint to convert.
+ * @returns The buffer representation of the bigint.
+ */
+export function bigintToBuffer(n: bigint): Buffer {
+    return beBigintToBuffer(n)
+}
+
+/**
+ * Converts a BigNumberish type to a buffer, handling various input types and converting them to bigint first if necessary.
+ * @param n The BigNumberish value to convert.
+ * @returns The buffer representation of the BigNumberish value.
+ */
+export function beBigNumberishToBuffer(n: BigNumberish): Buffer {
```

Comment:
These "bignumberish" helpers are another case where it's odd not to have LE alternatives.  Unlike the hex digits above, I do think this is a case where it's good to make the byte order explicit, since the use of a buffer means there is a choice.

Reply:
Makes sense! I've added `leBigNumberishToBuffer`, but not sure about `beBigNumberishToBigInt`.

Reply:
Why not?  I feel like there should be symmetry.

Reply:
Right, a bignumber-ish could also be a buffer indeed.

Reply:
The case where the BigNumberish is itself a buffer is a tricky one here.  It raises the question of whether the BE/LE prefix applies to the input buffer, or the output buffer, or both.  I think it's too complicated to introduce 4 separate options to separately specify input and output byte order.  So I could see two approaches which could be consistent here:

1) The LE/BE specifier applies to **both** input and output.  That means when a conversion is needed, use the selected byte order.  But if the input is already a buffer, it's assumed to already be in the desired byte order, so it can be passed through unchanged.
2) The LE/BE specifier applies to the "Buffer" output.  BigNumberish is assumed (and should be documented) to always assume a BE buffer.  In this case, if the input is a Buffer you'd need to reverse it as you say.

I think #1 is the simplest option for this function.  It's really an argument justifying how it already works.  I think #2 might be more consistent broadly, because all the functions which accept a BigNumberish are doing BE conversion by default.

Realizing this complexity makes me feel more like it's best to not provide a LE version after all.  I'd just leave the function as-is, and document what it does for clarity.  In particular, document that if the input is a buffer it's left unmodified, and if the input is not a buffer it's converted using BE order.  So long as callers know that, they can figure out what they need to do.

Reply:
> Realizing this complexity makes me feel more like it's best to not provide a LE version after all. I'd just leave the function as-is, and document what it does for clarity.

I agree. I removed the LE version and added more documentation to explain how they work.